novawallet/Modules/AssetDetails/AssetDetailsViewFactory.swift

```diff
@@ -5,7 +5,8 @@ struct AssetDetailsViewFactory {
     static func createView(
         assetListObservable: AssetListModelObservable,
         chain: ChainModel,
-        asset: AssetModel
+        asset: AssetModel,
+        swapCompletionClosure: SwapCompletionClosure?
```

Comment:
I would wrap assetListObservable and swapCompletionClosure into struct AssetOperationState